### ISSUE TITLE
"de" obl or obj complement of the adverbs "afin", "avant", "lors"

### ISSUE THREAD

**Issue by perrier54** [surfacesyntacticud/SUD_French-GSD#10]
When the adverb "avant" have as a complement an infinitive introduced by the preposition "de", the dependency "avant" -> "de" is a `comp:obj` dependency , because the complement can be replaced by a noun phrase.

> Example: _avant de partir_ (before he leaves) - _avant son départ_ (before his departure)

For "lors" and "afin" governing the preposition "de", we also choose  a `comp:obj' dependency, although there is no obvious justification.

**Comment by bguil** [comment]
GSD used `comp:obj` in all cases: https://universal.grew.fr/?custom=695a2cd25301f
@sylvainkahane: do you agree to apply the same to other treebanks?

**Comment by sylvainkahane** [comment]
Yes, if we have to choose between comp:obj and comp:obl, comp:obj is better. But I wonder if it won't be better to only use comp with adposition and to reserve comp:obj for objects of verbs. We often need to query the verb objects and it is confusing to have both verb object and adposition complements.
